From 156479dac2e577ae6d38ee630f349a2742c48aba Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <megi@xff.cz>
Date: Wed, 14 Jun 2023 00:48:21 +0200
Subject: pci: Workaround ITS timeouts on poweroff/reboot on Orange Pi 5 Plus

Call to pci_free_irq_vectors on PCIE2x ports causes ITS timeout messages
and delayed shutdown/reboot (by up to a minute).

The root cause will be elsewhere. This is not for upstream. Anyway,
there should be no harm from this, since we're shutting down anyway.

Signed-off-by: Ondrej Jirman <megi@xff.cz>
---
 drivers/pci/pcie/portdrv.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/pci/pcie/portdrv.c b/drivers/pci/pcie/portdrv.c
index 6af5e0425872..5ba7da8fcde5 100644
--- a/drivers/pci/pcie/portdrv.c
+++ b/drivers/pci/pcie/portdrv.c
@@ -739,7 +739,7 @@ static void pcie_portdrv_shutdown(struct pci_dev *dev)
 		pm_runtime_dont_use_autosuspend(&dev->dev);
 	}
 
-	pcie_port_device_remove(dev);
+	device_for_each_child(&dev->dev, NULL, remove_iter);
 }
 
 static pci_ers_result_t pcie_portdrv_error_detected(struct pci_dev *dev,
-- 
2.35.3

